frontend/public/components/secrets/create-secret/SecretFormWrapper.tsx

```diff
@@ -40,11 +40,14 @@ export const SecretFormWrapper: React.FC<BaseEditSecretProps_> = (props) => {
   const [secret, setSecret] = React.useState(initialSecret);
   const [inProgress, setInProgress] = React.useState(false);
   const [error, setError] = React.useState();
-  const [stringData, setStringData] = React.useState(
-    _.mapValues(_.get(props.obj, 'data'), (value) => {
-      return value ? Base64.decode(value) : '';
-    }),
-  );
+  const [stringData, setStringData] = React.useState({});
+
+  React.useEffect(() => {
+    setStringData(
+      _.mapValues(_.get(props.obj, 'data'), (value) => (value ? Base64.decode(value) : '')),
+    );
+  }, [props.obj]);
```

Comment:
I don't think we want this. It will cause data loss if the secret updates on the server side while the user is editing.

Reply:
Thats probably a right call. The reason behind this addition was that the `stringData` was being set to `{}` when trying to edit the _Image Pull Secret_ secret type even though the contents of `props.obj` remained the same. What this meant is that whenever we wanted to edit a secret of this type the input fields of the edit form were blank (see vid). 

Now I figured that the the stringData was being set to `{}` in the [onDataChanged](https://github.com/openshift/console/blob/5d88bd5a921463c92b460bab8247bd050af5124f/frontend/public/components/secrets/create-secret/SecretFormWrapper.tsx#L54), due to it being called inside the child components with only the `base64StringData` attribute of `SecretChangeData`.

What's weird is that in your PR your`PullSecretCredentialsForm` you similarly only [set](https://github.com/openshift/console/pull/14490/files#diff-e7eae5a321688edf91bcc50f37b43cc1bb741a03c6f3940802fb5a2e706cdfafR29) the `stringData` in the `onChange` callback and the tests seems to be passing.


https://github.com/user-attachments/assets/22a31afe-b5eb-4071-b563-f9cf63043823